File: auth-application/src/main/java/com/example/auth/application/port/in/AssignRoleUseCase.java
package com.example.auth.application.port.in;

import com.example.auth.domain.common.TenantId;
import com.example.auth.domain.common.UserId;
import java.util.UUID;

public interface AssignRoleUseCase {

    void assign(Command cmd);

    record Command(
            TenantId tenantId,
            UserId targetUserId,
            UUID roleId,
            UserId actorUserId,
            String ipAddress) {
    }
}

File: auth-application/src/main/kotlin/com/example/auth/application/port/in/AssignRoleUseCase.kt
package com.example.auth.application.port.`in`

import com.example.auth.domain.common.TenantId
import com.example.auth.domain.common.UserId
import java.util.UUID

interface AssignRoleUseCase {

    fun assign(cmd: Command)

    @JvmRecord
    data class Command(
        val tenantId: TenantId,
        val targetUserId: UserId,
        val roleId: UUID,
        val actorUserId: UserId,
        val ipAddress: String?,
    )
}
